### PR TITLE
Minchingtonak/phoenix migration

### PR DESCRIPTION
**Pull request by minchingtonak**

### CODE REVIEW

File: src/org/hammerhead226/sharkmacro/Parser.java [hunk 1]
```diff
@@ -224,4 +214,24 @@ protected static int findLatestNumberedFile(String directory, String prefix) {
...
+			try {
+				Files.createDirectories(p);
+				System.out.println("Created Directory: " + directory);
+			} catch (IOException e) {
+				e.printStackTrace();
+				return false;
+			}
+		}
+		return true;
+	}
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/Parser.java]: I think `check` could be named better, maybe something like `enforce` or `touch` ([Unix](https://en.wikipedia.org/wiki/Touch_(Unix))).

File: src/org/hammerhead226/sharkmacro/Parser.java [hunk 2]
```diff
@@ -81,19 +81,7 @@ public Parser(String directory, String prefix, String filename) {
...
-		if (Files.notExists(p)) {
-			try {
-				Files.createDirectories(p);
-				System.out.println("Created Directory: " + directory);
-			} catch (IOException e) {
-				e.printStackTrace();
-				return false;
-			}
-		}
-
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/Parser.java]: I think a call to `checkDirectory()` is still useful here.

File: src/org/hammerhead226/sharkmacro/motionprofiles/Profile.java [hunk 3]
```diff
@@ -121,8 +124,16 @@ public void execute(int leftGainsProfile, int rightGainsProfile) {
 	 * profile's execution is interrupted.
 	 */
 	public void onInterrupt() {
-		left.onInterrupt();
-		right.onInterrupt();
+		if (left != null) {
+			left.onInterrupt();
+		} else {
+			DriverStation.getInstance().reportError("Tried to interrupt null instance of left ProfileHandler!", false);
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/Profile.java]: "Tried" is technically not right, maybe something like "No left instance of ProfileHandler to interrupt". Also, probably a warning at strongest, not an error.

File: src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java [hunk 4]
```diff
@@ -155,9 +168,9 @@ public void manage() {
 	 * @param t
 	 *            the {@link TalonState} to set the Talon to
 	 */
-	private void setMode(TalonState t) {
-		this.currentMode = t;
-		talon.set(t.getValue());
+	private void setMode(SetValueMotionProfile v) {
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java]: Would be happier with a more descriptive parameter name here.

File: src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java [hunk 5]
```diff
@@ -244,49 +279,6 @@ public void run() {
...
-	 *            the integer value of the {@link TalonState}
-	 */
-	private TalonState(int value) {
-		this.value = value;
-	}
-
-	/**
-	 * @return The integer value of this {@link TalonState}
-	 */
-	public int getValue() {
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java]: Java can have methods in enums? WAT.

File: src/org/hammerhead226/sharkmacro/motionprofiles/Recording.java [hunk 6]
```diff
@@ -116,4 +133,30 @@ private boolean areEqual(int idx, double comparator) {
...
+	 * 
+	 * @param rawUnits
+	 *            raw sensor units
+	 * @param sensorUnitsPerRotation
+	 *            amount of raw units per wheel rotation
+	 * @return raw units as wheel rotations
+	 */
+	private double toRotations(double rawUnits, double sensorUnitsPerRotation) {
+		return rawUnits * (1.0 / sensorUnitsPerRotation);
+	}
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/Recording.java]: Would it be useful to provide a `sensorUnitsPerRotation = 4096` version of the method?

File: src/org/hammerhead226/sharkmacro/motionprofiles/Recording.java [hunk 7]
```diff
@@ -116,4 +133,30 @@ private boolean areEqual(int idx, double comparator) {
...
+	 * 
+	 * @param rawUnitsPer100ms
+	 *            raw sensor units
+	 * @param sensorUnitsPerRotation
+	 *            amount of raw units per wheel rotation
+	 * @return raw units per 100ms as RPM
+	 */
+	private double toRPM(double rawUnitsPer100ms, double sensorUnitsPerRotation) {
+		return rawUnitsPer100ms * (600.0 / sensorUnitsPerRotation);
+	}
```
Comment (emmercm) [inline src/org/hammerhead226/sharkmacro/motionprofiles/Recording.java]: Same question as `toRotations()`.

File: src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java (hunk 5 above)
Comment (minchingtonak) [reply src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java]: Best programming language

**Pull request merged**